### ISSUE TITLE
Golint wants to elide type that can be inferred, but this breaks docs

### ISSUE THREAD

**Issue by SamWhited** [golang/lint#307]
I found a number of closed issues about golint wanting to elide types that couldn't actually be inferred, but I'd like to dispute that this warning is necessary; it can actually be harmful (eg. it's not just a matter of style). For example, consider the following API:

```go
package foo

type Foo string

var (
    // Bar is some common, predeclared Foo.
    Bar Foo = Foo("one")

    // Baz is some other common Foo.
    Baz Foo = Foo("two")   
)
```

If we `godoc foo` against this package, our common Foo vars are grouped under the Foo type. However, `golint` complains about the unnecessary type, but if we remove it to make golint happy, then the list of common Foos appears in a vars section at the top of the document which may be nowhere near Foo in a large package, and is very confusing.

Maybe `golint` could warn about *mixed* inferred types (probably only in lists where all the RHS types were the same) instead, so it would not warn on the above, but would warn on:

```go
var (
    // Bar is some common, predeclared Foo.
    Bar Foo = Foo("one")

    // Baz is some other common Foo.
    Baz = Foo("two")   
)
```

An experimental implementation of this idea can be found in 1426a3b4d43c2ff45dc5a1cc2f3d24c846c89c0f

**Comment by SamWhited** [comment]
Oops, made a duplicate somehow. Closing this as a duplicate of #429
